Sources/Helpers/OASvgHelper.mm

```diff
@@ -29,37 +39,72 @@ + (nullable UIImage *) mapImageNamed:(NSString *)name scale:(float)scale
     return [self.class mapImageFromSvgResource:name width:scaledSize height:scaledSize];
 }
 
-+ (nullable UIImage *) imageNamed:(NSString *)path
++ (nullable UIImage *)imageNamed:(NSString *)path
 {
     NSString *resourceName = [path lastPathComponent];
-    NSString *subpath = [path stringByDeletingLastPathComponent];
-    const auto resourcePath = [[NSBundle mainBundle] pathForResource:resourceName
-                                                              ofType:@"svg"
-                                                         inDirectory:subpath];
-    if (resourcePath == nil)
-        return nil;
-
-    CGFloat scaleFactor = [[UIScreen mainScreen] scale];
-    float scaledSize = kDefaultIconSize * scaleFactor;
+    NSString *resourcePath;
+    {
+        QReadLocker scopedLocker(&iconsCacheLock);
+        if ([resourcesNotExistCache containsObject:resourceName])
+            return nil;
+        resourcePath = resourcesExistCache[resourceName];
+    }
+    if (!resourcePath)
+    {
+        resourcePath = [[NSBundle mainBundle] pathForResource:resourceName
+                                                       ofType:@"svg"
+                                                  inDirectory:[path stringByDeletingLastPathComponent]];
+        {
+            QWriteLocker scopedLocker(&iconsCacheLock);
+            if (!resourcePath)
+            {
+                [resourcesNotExistCache addObject:resourceName];
+                return nil;
+            }
+            resourcesExistCache[resourceName] = resourcePath;
+        }
+    }
+    float scaledSize = kDefaultIconSize * UIScreen.mainScreen.scale;
     return [OANativeUtilities skImageToUIImage:[OANativeUtilities skImageFromSvgResourcePath:resourcePath width:scaledSize height:scaledSize]];
 }
 
-+ (UIImage *) mapImageFromSvgResource:(NSString *)resourceName width:(float)width height:(float)height
++ (nullable NSString *)mapImagePathFromSvgResource:(NSString *)resourceName
```

Comment:
Create single method for operating with cache like `(nullable NSString *)mapImageResourcePathNamed:(NSString *)resourceName resourceDir:(NSString *)resourceDir`